src/main/java/com/minecolonies/coremod/entity/ai/citizen/smelter/EntityAIWorkSmelter.java

```diff
@@ -371,194 +244,24 @@ private void gatherOreFromFurnaceWithChance(final TileEntityFurnace furnace)
-            {
-                if (!InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel))
-                {
-                    walkTo = null;
-                    return SMELTER_GET_FIREWOOD;
-                }
-
-                InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoInItemHandler(
-                        new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel, STACKSIZE,
-                        new InvWrapper((TileEntityFurnace) entity), FUEL_SLOT);
-            }
-
-            walkTo = null;
-            return START_WORKING;
-        }
-        walkTo = null;
-        setDelay(STANDARD_DELAY);
-        return SMELTER_SMELT_ORE;
-    }
-
-    private BlockPos getPositionOfOvenToRetrieveFrom()
-    {
-        for (final BlockPos pos : ((BuildingSmeltery) getOwnBuilding()).getFurnaces())
-        {
-            final TileEntity entity = world.getTileEntity(pos);
-            if (entity instanceof TileEntityFurnace && !((TileEntityFurnace) entity).isBurning()
-                    && (!ItemStackUtils.isEmpty(((TileEntityFurnace) entity)
-                    .getStackInSlot(RESULT_SLOT))
-                    || !ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(ORE_SLOT))))
-            {
-                worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.retrieving"));
-                return pos;
-            }
-        }
-        return null;
-    }
-
-    @Override
-    protected int getActionsDoneUntilDumping()
-    {
-        return 1;
-    }
-
-    private AIState startWorking()
-    {
-        final BlockPos posOfOven = getPositionOfOvenToRetrieveFrom();
-        if (posOfOven != null)
-        {
-            walkTo = posOfOven;
-            return SMELTER_RETRIEVE_ORE;
-        }
-
-        final int amountOfOre = InventoryUtils.getItemCountInProvider(getOwnBuilding(), EntityAIWorkSmelter::isSmeltableOre)
-                + InventoryUtils.getItemCountInItemHandler(new InvWrapper(worker.getInventoryCitizen()), EntityAIWorkSmelter::isSmeltableOre);
-
-        if (amountOfOre <= 0)
-        {
-            worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.gathering"));
-            if (!getOwnBuilding().hasWorkerOpenRequestsOfType(worker.getCitizenData(), TypeToken.of(SmeltableOre.class)))
-            {
-                worker.getCitizenData().createRequestAsync(new SmeltableOre(STACKSIZE));
-            }
-        }
-
-        return checkForAdditionalJobs(amountOfOre);
-    }
-
     /**
      * If no clear tasks are given, check if something else is to do.
-     * @param amountOfOre the amount of ore.
      * @return the next AIState to traverse to.
      */
-    private AIState checkForAdditionalJobs(final int amountOfOre)
+    protected AIState checkForAdditionalJobs()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1161)